### PR TITLE
Committing a basic usage notebook

### PR DESCRIPTION
**Pull request by ericmjl**
This was partially a tutorial for myself, and also a way to introduce a "simple complex example" for PyMC3 users.

Also discovered some UI/UX limitations, and will be posting them shortly.

### CODE REVIEW

**Comment by lucianopaz** [comment]
Nice! Thanks @ericmj! I would wait for #180 so you don't have to plot the log off the scale. One other thing, could you try to add a cell on posterior predictive and prior predictive sampling?

**Comment by lucianopaz** [comment]
One more super small nitpick that could cause confusion further down the line, `yield` doesn't return a probability distribution, it returns a value that is drawn or is assumed to come from the probability distribution.

You can try this:
```python
@pm.model
def model():
    dist = pm.Normal("x", 0, 1)
    x = yield dist
    print(dist)
    print(x)

_, state = pm.evaluate_model(model())
```

You'll see that it will print out that `dist` is our `Normal` distribution instance while `x` will be a tensor of values. Using the trick of not yielding from the distribution immediately, you can modify the distribution inside the model's function before you draw samples from it. For example, you could collect the distribution's `log_prob` in a deterministic like this:

```python
@pm.model
def model():
    dist = pm.Normal("x", 0, 1)
    x = yield dist
    logp = pm.Deterministic("logp", dist.log_prob(x))

_, state = pm.evaluate_model(model())
```

You can also play with the distribution's shape and I imagine that you could try many other things that don't come to my mind now.

I know that what I'm saying isn't basic usage and shouldn't be in this guide that you're writing, but I think that it's important not to mix the terms distribution, random variable and samples from a distribution.

**Comment by ferrine** [comment]
<p>I'm interested in untransformed sigma, it should be visible in the chain. If not, it is a case we should fix that</p>

File: notebooks/basic-usage.ipynb [hunk 1]
```diff
@@ -0,0 +1,310 @@
+{
```
Comment (twiecki) [inline notebooks/basic-usage.ipynb]: <p>Should always specify whether we talk about PyMC3 or PyMC4 here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc4/pull/190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (twiecki) [inline notebooks/basic-usage.ipynb]: <p>I think we should start putting the <code>name</code> kwarg as the last one everyone.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc4/pull/190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (twiecki) [inline notebooks/basic-usage.ipynb]: <p>I think we moved this to the top namespace now.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc4/pull/190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (twiecki) [inline notebooks/basic-usage.ipynb]: <p>We should probably change that behavior and just return an arviz trace object with no reference to the tensors.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc4/pull/190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (twiecki) [inline notebooks/basic-usage.ipynb]: <p>if you add a <code>;</code> at the end you suppress the weird plot return value being printed in the ouptut.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc4/pull/190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Comment by lucianopaz** [comment]
> 
> 
> I'm interested in untransformed sigma, it should be visible in the chain. If not, it is a case we should fix that

@ferrine, it's currently an open issue (#169) and @ColCarroll was working on a PR to fix it (#180)

File: notebooks/basic-usage.ipynb (hunk 1 above)
Comment (ericmjl) [reply notebooks/basic-usage.ipynb]: <p>@twiecki I think I have made that distinction clear here?</p>

Comment (ericmjl) [reply notebooks/basic-usage.ipynb]: <p>Yeah, my bad. I have used pm.sample correctly, but will remove the import.</p>

Comment (ericmjl) [reply notebooks/basic-usage.ipynb]: <p>That probably should be a later PR, out of scope for this one?</p>

Comment (ericmjl) [reply notebooks/basic-usage.ipynb]: <p>:thumbsup: (I wish there were emojis here)</p>

**Comment by ericmjl** [comment]
> One more super small nitpick that could cause confusion further down the line, yield doesn't return a probability distribution, it returns a value that is drawn or is assumed to come from the probability distribution.

Cool, thanks for the catch! I was definitely being sloppy with terminology there.

File: notebooks/basic-usage.ipynb (hunk 1 above)
Comment (ericmjl) [reply notebooks/basic-usage.ipynb]: Done!

**Comment by ericmjl** [comment]
> In other words, we should explain that a pm.model is a program that defers its computation and returns control flow back to the PyMC inference engine every time we yield. Perhaps this explanation belongs in a separate notebook?

@eigenfoo, I think @ferrine or someone else explained what's going on in the PyMC design guide notebook - happy to be corrected on this one.

**Comment by ericmjl** [comment]
Any further comments on the notebook? (I think I've addressed all of them.) If not, I'll hit the big green button tonight (Boston time).

**Comment by twiecki** [comment]
Hit it!

File: notebooks/basic-usage.ipynb [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
```
Comment (aloctavodia) [inline notebooks/basic-usage.ipynb]: <p>Probably "sampling stats" is a better description than "diagnostics"</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc4/pull/190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (aloctavodia) [inline notebooks/basic-usage.ipynb]: <p>nitpick, what about using a number of samples other than 1000, just to get a more visual understanding of the meaning of each dimension?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc4/pull/190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

**Comment by lucianopaz** [comment]
@ericmjl, I would wait for #180 to get merged before merging this PR. Once #180 gets in, you wouldn't have to plot the transformed scale, and you would be able to plot the standard `HalfNormal`'s posterior.

**Comment by ericmjl** [comment]
@lucianopaz I’m going to merge this first, so that I can get this out of my headspace, but I’ll also raise an appropriate issue to remind myself to fix this nb once #180 is done.

**Pull request merged**